server/src/grpc/server.rs

```diff
@@ -58,23 +73,66 @@ pub async fn grpc_server(config: Config) {
         }
     };
 
+    let imp = ServerImpl::default();
     let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
-    let imp = GRPCServerImpl::default();
     health_reporter
-        .set_serving::<RpcServiceServer<GRPCServerImpl>>()
+        .set_serving::<RpcServiceServer<ServerImpl>>()
         .await;
 
     //start server
-    grpc_info!("Starting GRPC servers on: {}.", full_grpc_addr);
+    grpc_info!("Starting gRPC services on: {}", full_grpc_addr);
     match Server::builder()
         .add_service(health_service)
         .add_service(RpcServiceServer::new(imp))
-        .serve_with_shutdown(full_grpc_addr, shutdown_signal("grpc"))
+        .serve_with_shutdown(full_grpc_addr, shutdown_signal("grpc", shutdown_rx))
         .await
     {
-        Ok(_) => grpc_info!("gRPC server running at: {}.", full_grpc_addr),
+        Ok(_) => grpc_info!("gRPC server running at: {}", full_grpc_addr),
         Err(e) => {
-            grpc_error!("could not start gRPC server: {}", e);
+            grpc_error!("Could not start gRPC server: {}", e);
         }
     };
 }
+
+#[cfg(feature = "stub_server")]
+#[tonic::async_trait]
+impl RpcService for ServerImpl {
+    async fn is_ready(
+        &self,
+        request: Request<ReadyRequest>,
+    ) -> Result<Response<ReadyResponse>, Status> {
+        grpc_warn!("(MOCK) contact server.");
+        grpc_debug!("(MOCK) [{:?}].", request);
+        let response = ReadyResponse { ready: true };
+        Ok(Response::new(response))
+    }
+
+    async fn cargo_confirmation(
+        &self,
+        request: Request<CargoConfirmationRequest>,
+    ) -> Result<Response<CargoConfirmationResponse>, Status> {
+        grpc_warn!("(MOCK) contact server.");
+        grpc_debug!("(MOCK) [{:?}].", request);
+        let response = CargoConfirmationResponse { success: true };
+        Ok(Response::new(response))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn test_grpc_server_is_ready() {
+        lib_common::logger::get_log_handle().await;
+        ut_info!("Start.");
+
+        let imp = ServerImpl::default();
+        let result = imp.is_ready(Request::new(ReadyRequest {})).await;
+        assert!(result.is_ok());
+        let result: ReadyResponse = result.unwrap().into_inner();
+        assert_eq!(result.ready, true);
+
+        ut_info!("Success.");
+    }
```

Comment:
We can also add a test for the cargo confirmation:
```rust
    #[tokio::test]
    async fn test_grpc_cargo_confirmation() {
        lib_common::logger::get_log_handle().await;
        ut_info!("start");

        let imp = ServerImpl::default();
        let result = imp
            .cargo_confirmation(Request::new(CargoConfirmationRequest {
                itinerary_id: String::from(""),
                parcel_id: String::from(""),
            }))
            .await;
        assert!(result.is_ok());
        let result: CargoConfirmationResponse = result.unwrap().into_inner();
        assert!(result.success);

        ut_info!("success");
    }
    ```

Reply:
https://github.com/aetheric-oss/svc-contact/pull/11/commits/59b2d52934224723bfa2ae6be1b62833f0e9a68a